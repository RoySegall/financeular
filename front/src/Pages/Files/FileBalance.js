import CardTable from "../../Components/Table/CardTable";
import Card from "../../Components/Card/Card";
import React from "react";
import {formatToCurrency} from "./File.service";

export default ({
  incomes, expenses, totalIncomes, totalExpenses,
  infoBoxIcon, infoBoxColor, infoBoxTitle, balance,
  incomeCurrentPage, setIncomeCurrentPage, expensesCurrentPage, setExpensesCurrentPage
}) => <section className="pt-4 flex">
  <div className="mr-10 w-3/6">

    <CardTable
      title="Incomes"
      headers={['Title', 'Value']}
      rows={incomes}
      perPage={5}
      currentPage={incomeCurrentPage}
      setCurrentPage={setIncomeCurrentPage} />

    <div>
      <Card
        icon={infoBoxIcon}
        iconType={infoBoxColor}
        title={infoBoxTitle}
        className={"w-full"}
        text={
          <>
            <p>
              All your income(s) are summed to <b>{formatToCurrency(totalIncomes)}</b> while the
              expenses are summed to <b>{formatToCurrency(totalExpenses)}</b>.
            </p>
            <p>
              The balance stands on <b>{formatToCurrency(balance)}</b>.
            </p>
          </>} />
    </div>
  </div>

  <div className="w-3/6">
    <CardTable
      title="Expenses"
      headers={['Title', 'Value', 'Date']}
      rows={expenses}
      perPage={10}
      currentPage={expensesCurrentPage}
      setCurrentPage={setExpensesCurrentPage} />
  </div>

</section>