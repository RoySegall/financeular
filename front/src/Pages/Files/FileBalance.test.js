import renderer from "react-test-renderer";
import React from "react";
import FileBalance from "./FileBalance";

describe('File balance component', () => {

  it('Validate the content for over draft', () => {
    const incomes = [
      ['income', 1200],
      ['income', 1200],
      ['income', 1200],
      ['income', 1200],
      ['income', 1200],
      ['income', 1200],
      ['income', 1200],
      ['income', 1200],
    ];
    const expenses = [
      ['expense', 1200, '01-02-1990'],
      ['expense', 1200, '01-02-1990'],
      ['expense', 1200, '01-02-1990'],
      ['expense', 1200, '01-02-1990'],
      ['expense', 1200, '01-02-1990'],
      ['expense', 1200, '01-02-1990'],
      ['expense', 1200, '01-02-1990'],
      ['expense', 1200, '01-02-1990'],
      ['expense', 1200, '01-02-1990'],
      ['expense', 1200, '01-02-1990'],
      ['expense', 1200, '01-02-1990'],
      ['expense', 1200, '01-02-1990'],
      ['expense', 1200, '01-02-1990'],
      ['expense', 1200, '01-02-1990'],
      ['expense', 1200, '01-02-1990'],
    ];

    const tree = renderer
      .create(<FileBalance {...{
        incomes, expenses,
        infoBoxIcon: <>Box icon</>,
        infoBoxColor: 'green',
        infoBoxTitle: 'awesome',
        totalIncomes: 1234,
        totalExpenses: 5678,
        balance: 9000,
        incomeCurrentPage: 0,
        setIncomeCurrentPage: () => {},
        expensesCurrentPage: () => {},
        setExpensesCurrentPage: () => {},
      }}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});