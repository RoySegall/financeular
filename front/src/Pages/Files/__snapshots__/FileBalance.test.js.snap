// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File balance component Validate the content for over draft 1`] = `
<section
  className="pt-4 flex"
>
  <div
    className="mr-10 w-3/6"
  >
    <div
      className="relative flex flex-col min-w-0 break-words w-full mb-6 shadow-lg rounded bg-white undefined"
    >
      <div
        className="rounded-t mb-0 px-4 py-3 border-0"
      >
        <div
          className="flex flex-wrap items-center"
        >
          <div
            className="relative w-full px-4 max-w-full flex-grow flex text-left flex justify-between"
          >
            <h3
              className="font-semibold text-lg text-gray-800"
            >
              Incomes
            </h3>
            <span>
              <button
                className="pr-1 cursor-pointer underline"
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-angle-double-left fa-w-14 "
                  data-icon="angle-double-left"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </button>
              1
               of 
              2
              <button
                className="pl-1 cursor-pointer underline"
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-angle-double-right fa-w-14 "
                  data-icon="angle-double-right"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </button>
            </span>
          </div>
        </div>
      </div>
      <div
        className="block w-full overflow-x-auto"
      >
        <table
          className="items-center w-full bg-transparent border-collapse"
        >
          <thead>
            <tr>
              <th
                className="px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-no-wrap font-semibold text-left bg-gray-100 text-gray-600 border-gray-200"
              >
                Title
              </th>
              <th
                className="px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-no-wrap font-semibold text-left bg-gray-100 text-gray-600 border-gray-200"
              >
                Value
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-no-wrap p-4 text-left"
                id={0}
              >
                income
              </td>
              <td
                className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-no-wrap p-4 text-left"
                id={1}
              >
                1200
              </td>
            </tr>
            <tr>
              <td
                className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-no-wrap p-4 text-left"
                id={0}
              >
                income
              </td>
              <td
                className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-no-wrap p-4 text-left"
                id={1}
              >
                1200
              </td>
            </tr>
            <tr>
              <td
                className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-no-wrap p-4 text-left"
                id={0}
              >
                income
              </td>
              <td
                className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-no-wrap p-4 text-left"
                id={1}
              >
                1200
              </td>
            </tr>
            <tr>
              <td
                className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-no-wrap p-4 text-left"
                id={0}
              >
                income
              </td>
              <td
                className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-no-wrap p-4 text-left"
                id={1}
              >
                1200
              </td>
            </tr>
            <tr>
              <td
                className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-no-wrap p-4 text-left"
                id={0}
              >
                income
              </td>
              <td
                className="border-t-0 px-6 align-middle border-l-0 border-r-0 text-xs whitespace-no-wrap p-4 text-left"
                id={1}
              >
                1200
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div>
      <section
        className="align-middle bg-white w-9/12 m-auto m-0 relative flex flex-col mb-6 shadow-lg rounded p-2 px-4 py-5 w-full"
      >
        <div
          className="m-auto m-0 p-3 text-center inline-flex items-center justify-center w-14 h-14 mb-5 shadow-lg rounded-full text-2xl bg-green-400 text-white"
        >
          Box icon
        </div>
        <h2
          className="text-xl font-semibold text-black pb-4"
        >
          awesome
        </h2>
        <p
          className=" mt-2 mb-4 text-gray-600"
        >
          <p>
            All your income(s) are summed to 
            <b>
              1,234
            </b>
             while the expenses are summed to 
            <b>
              5,678
            </b>
            .
          </p>
          <p>
            The balance stands on 
            <b>
              9,000
            </b>
            .
          </p>
        </p>
      </section>
    </div>
  </div>
  <div
    className="w-3/6"
  >
    <div
      className="relative flex flex-col min-w-0 break-words w-full mb-6 shadow-lg rounded bg-white undefined"
    >
      <div
        className="rounded-t mb-0 px-4 py-3 border-0"
      >
        <div
          className="flex flex-wrap items-center"
        >
          <div
            className="relative w-full px-4 max-w-full flex-grow flex text-left flex justify-between"
          >
            <h3
              className="font-semibold text-lg text-gray-800"
            >
              Expenses
            </h3>
            <span>
              <button
                className="pr-1 cursor-pointer underline"
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-angle-double-left fa-w-14 "
                  data-icon="angle-double-left"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </button>
              () =&gt; {}1
               of 
              2
              <button
                className="pl-1 cursor-pointer underline"
                onClick={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-angle-double-right fa-w-14 "
                  data-icon="angle-double-right"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  style={Object {}}
                  viewBox="0 0 448 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
                    fill="currentColor"
                    style={Object {}}
                  />
                </svg>
              </button>
            </span>
          </div>
        </div>
      </div>
      <div
        className="block w-full overflow-x-auto"
      >
        <table
          className="items-center w-full bg-transparent border-collapse"
        >
          <thead>
            <tr>
              <th
                className="px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-no-wrap font-semibold text-left bg-gray-100 text-gray-600 border-gray-200"
              >
                Title
              </th>
              <th
                className="px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-no-wrap font-semibold text-left bg-gray-100 text-gray-600 border-gray-200"
              >
                Value
              </th>
              <th
                className="px-6 align-middle border border-solid py-3 text-xs uppercase border-l-0 border-r-0 whitespace-no-wrap font-semibold text-left bg-gray-100 text-gray-600 border-gray-200"
              >
                Date
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </div>
</section>
`;
